We need to disable these three tests.

- JointTest.test_nested_distributions__with_device
- JointTest.test_nested_distributions__without_device
- JointTest.test_nested_distributions__without_jit
<details><summary>error log on aarch64-linux</summary>

```
=================================== FAILURES ===================================
_______________ JointTest.test_nested_distributions__with_device _______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__with_device>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
_____________ JointTest.test_nested_distributions__without_device ______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__without_device>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
_______________ JointTest.test_nested_distributions__without_jit _______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__without_jit>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
```
</details>

--- a/pkgs/development/python-modules/distrax/default.nix
+++ b/pkgs/development/python-modules/distrax/default.nix
@@ -1,21 +1,27 @@
 { lib
-, fetchPypi
 , buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, chex
+, jaxlib
 , numpy
 , tensorflow-probability
-, chex
 , dm-haiku
 , pytestCheckHook
-, jaxlib
 }:
 
 buildPythonPackage rec {
   pname = "distrax";
-  version = "0.1.4";
+  version = "0.1.5";
+  pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-klXT5wfnWUGMrf5sQhYqz7Foc/Ou5y4GIFgtTff1ZFQ=";
+  disabled = pythonOlder "3.9";
+
+  src = fetchFromGitHub {
+    owner = "google-deepmind";
+    repo = "distrax";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-A1aCL/I89Blg9sNmIWQru4QJteUTN6+bhgrEJPmCrM0=";
   };
 
   buildInputs = [
@@ -32,6 +38,26 @@ buildPythonPackage rec {
 
   pythonImportsCheck = [
     "distrax"
+  ];
+
+  disabledTests = [
+    # AssertionError on numerical values
+    # Reported upstream in https://github.com/google-deepmind/distrax/issues/267
+    "test_method_with_input_unnormalized_probs__with_device"
+    "test_method_with_input_unnormalized_probs__with_jit"
+    "test_method_with_input_unnormalized_probs__without_device"
+    "test_method_with_input_unnormalized_probs__without_jit"
+    "test_method_with_value_1d"
+    "test_nested_distributions__with_device"
+    "test_nested_distributions__without_device"
+    "test_nested_distributions__with_jit"
+    "test_nested_distributions__without_jit"
+    "test_stability__with_device"
+    "test_stability__with_jit"
+    "test_stability__without_device"
+    "test_stability__without_jit"
+    "test_von_mises_sample_gradient"
+    "test_von_mises_sample_moments"
   ];
 
   disabledTestPaths = [
@@ -54,7 +80,5 @@ buildPythonPackage rec {
     homepage = "https://github.com/deepmind/distrax";
     license = licenses.asl20;
     maintainers = with maintainers; [ onny ];
-    # Broken on all platforms (starting 2022-07-27)
-    broken = true;
   };
 }

